apps/api/src/app/blueprint/usecases/get-grouped-blueprints/get-grouped-blueprints.usecase.ts

```diff
@@ -14,14 +17,48 @@ export class GetGroupedBlueprints {
     builder: () => buildGroupedBlueprintsKey(),
     options: { ttl: WEEK_IN_SECONDS },
   })
-  async execute(): Promise<GroupedBlueprintResponse[]> {
-    const blueprints = await this.notificationTemplateRepository.findAllGroupedByCategory();
-    if (!blueprints) {
+  async execute(): Promise<GroupedBlueprintResponse> {
+    const groups = await this.fetchGroupedBlueprints();
+
+    const updatePopularBlueprints = this.updatePopularBlueprints(groups);
+
+    const popular = { name: POPULAR_TEMPLATES_GROUPED.name, blueprints: updatePopularBlueprints };
+
+    return { general: groups as GroupedBlueprint[], popular };
+  }
+
+  private async fetchGroupedBlueprints() {
+    const groups = await this.notificationTemplateRepository.findAllGroupedByCategory();
+    if (!groups) {
       throw new NotFoundException(
         `Blueprints for organization id ${NotificationTemplateRepository.getBlueprintOrganizationId()} were not found`
       );
     }
 
-    return blueprints;
+    return groups;
+  }
+
+  private extractStoredBlueprints(groupes: { name: string; blueprints: NotificationTemplateEntity[] }[]) {
+    return groupes
```

Comment:
```suggestion
  private extractStoredBlueprints(groups: { name: string; blueprints: NotificationTemplateEntity[] }[]) {
    return groups
```
Not french 😛

Reply:
Lol I had a lot of them no idea why, I renamed a lot, looks like I missed some 😅